src/components/code/CodeEditor.vue

```diff
@@ -1,16 +1,34 @@
 <template>
   <div class="code-container">
-    <span class="header">{{ type.toUpperCase() }}</span>
+    <span class="header">{{ type.toUpperCase() }}
+      <button @click="copyToClipBoard" class="copy-button" v-bind:id="type + 'SaveButton'">Copy</button>
+    </span>
     <pre><code><slot/></code></pre>
   </div>
 </template>
 
-<script setup>
+<script setup="props">
+
+import { useClipboard } from '@vueuse/core'
+
+const { copy } = useClipboard()
+
 export default {
   props: {
     type: { type: String, required: true },
+    code: { type: String, required: true },
   },
 }
+
+export function copyToClipBoard(){
+  document.getElementById(props.type + 'SaveButton').innerText="Copied!"
+  copy(props.code)
+  setTimeout(function(){
+    document.getElementById(props.type + 'SaveButton').innerText="Copy"
+  }, 2000);
```

Comment:
We could use a ref to avoid accessing the DOM with getElementById here.
In the template, you can replace `Copy` by `{{ copied ? 'Copied!' : 'Copy' }}`

The action could be written as
```
export function copyToClipboard() {
  copy(props.code)
}
```
And instead of a timeout, we can use the `text` ref that `useClipboard` from vueuse gives us to keep the `Copied!` label until the clipboard has that text
```
export const copied = computed( () => text.value === props.code )
```
If the code changes, or the the text in the clipboard changes, then the button is back to normal

What do you think?

Reply:
@rhythm1920 I think we can not access the `text.value` without prompting a request for access to the user, so we can not actually do the computed copied like I proposed, bummer.

```
export const copied = ref(0)
export function copyToClipboard() {
  copy(props.code)
  copied.value++
  setTimeout( () => {
    copied.value--
  }, 2000 )
}
```

Using a ref(counter) instead of a boolean for copied, allows users to press the copy button several times, without having the first timeout clear it.

Reply:
Ok I will work on it

Reply:
@matias-capeletto I am unable to understand what is `ref(0)` . Can you explain that?
I searched around the web and only found `$refs` and not a function like `ref()`.

Reply:
@rhythm1920 we are using Vue 3 reactivity (ref, computed, watch). You can read about them here: 
https://v3.vuejs.org/guide/composition-api-introduction.html#reactive-variables-with-ref
We are also using <script setup>, you can read more about it here: https://github.com/vuejs/rfcs/blob/sfc-improvements/active-rfcs/0000-sfc-script-setup.md

ref is reactive reference to a value. You access it using `r.value` in scripts, but can use it directly as `r` in templates. If you want that ref to be exposed to the template, you need to export it. You will need to import it first like:
```
<script setup>
import { ref } from 'vue'
export const copied = ref(0) // copied is a reactive ref, you can think of it as one piece of the old data options
</scipt>
```

Reply:
OK thanks a lot, I do not have much experience with vue so I was not knowing about `ref`

Reply:
I have made the required changes. Please review it.

Reply:
Great! Thanks for the changes 💯